### PR TITLE
NSA-2073 - allow invitation reactivation

### PR DESCRIPTION
**Pull request by bucky-uk**
Changes:
- Users that have been deactivated before accepting an invite can now be reactivated
- Warning banner added to profile summary of users with status === -2 (Deactivated Invitation)
- Actions menu also has a link to reactivate the invitation
- New API method added to directories client for handling reactivations.

### CODE REVIEW

File: src/infrastructure/directories/api.js [hunk 1]
```diff
@@ -252,6 +252,27 @@ const deactivateInvite = async (id, reason, correlationId) => {
...
+    await rp({
+      method: 'PATCH',
+      uri: `${config.directories.service.url}/invitations/${id.replace('inv-', '')}`,
+      headers: {
+        authorization: `bearer ${token}`,
+        'x-correlation-id': correlationId,
+      },
+      body: {
+        reason: reason,
+        deactivated: true,
```
Comment (bridgesr) [inline src/infrastructure/directories/api.js]: Think this needs to be deactivated: false

File: src/app/users/views/summaryPartial.ejs [hunk 2]
```diff
@@ -69,17 +84,22 @@
...
+                <% if (!locals.user.deactivated) { %>
+                    <li><a href="associate-organisation">Add organisation</a></li>
+                    <% if (locals.organisations.length > 0) { %>
+                        <li><a href="select-organisation">Add services</a></li>
+                    <% } %>
+                    <li><a href="edit-email">Change user's email address</a></li>
+                    <li><a href="confirm-invitation-deactivation">Deactivate user</a></li>
+                    <li><a href="resend-invitation"> Resend invitation email</a></li>
+                <% } %>
+                <% if (locals.user.deactivated) { %>
```
Comment (bridgesr) [inline src/app/users/views/summaryPartial.ejs]: This is only relevant for users so is not needed in this block, you can reactive in the banner which is fine

**Review by bridgesr — COMMENTED** [review]
It is worth checking whether there is a user that exists before resending the invitation email. You should be able to call directories `/users/{email}` to check if a user already exists for the email, this will stop people resending invitations from bookmarks etc.

File: src/infrastructure/directories/api.js (hunk 1 above)
Comment (bucky-uk) [reply src/infrastructure/directories/api.js]: Missed the commit... have updated and re-submitted

File: src/app/users/views/summaryPartial.ejs (hunk 2 above)
Comment (bucky-uk) [reply src/app/users/views/summaryPartial.ejs]: Removed and reverted back to original sidebar

**Review by floydprice — APPROVED** [review]

**Pull request merged**